bundles/org.openhab.binding.jablotron/src/main/java/org/openhab/binding/jablotron/internal/handler/JablotronJa100Handler.java

```diff
@@ -0,0 +1,203 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.jablotron.internal.handler;
+
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.binding.builder.ChannelBuilder;
+import org.eclipse.smarthome.core.thing.binding.builder.ThingBuilder;
+import org.eclipse.smarthome.core.thing.type.ChannelTypeUID;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.jablotron.internal.model.JablotronControlResponse;
+import org.openhab.binding.jablotron.internal.model.JablotronServiceDetailSegment;
+import org.openhab.binding.jablotron.internal.model.JablotronServiceDetailSegmentInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link JablotronJa100Handler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Ondrej Pecta - Initial contribution
+ */
+@NonNullByDefault
+public class JablotronJa100Handler extends JablotronAlarmHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(JablotronJa100Handler.class);
+
+    public JablotronJa100Handler(Thing thing, String alarmName) {
+        super(thing, alarmName);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (channelUID.getId().startsWith("STATE_") && command instanceof StringType) {
+            scheduler.execute(() -> {
+                controlSTATESection(channelUID.getId(), command.toString());
+            });
+        }
+
+        if (channelUID.getId().startsWith("PGM_") && command instanceof OnOffType) {
+            scheduler.execute(() -> {
+                controlPGMSection(channelUID.getId(), command.equals(OnOffType.ON) ? "set" : "unset");
+            });
+        }
+    }
+
+    private void createChannel(JablotronServiceDetailSegment section) {
+        if (section.getSegmentId().startsWith("PGM_")) {
+            createPGMChannel(section.getSegmentId(), section.getSegmentName());
+        } else {
+            createStateChannel(section.getSegmentId(), section.getSegmentName());
+        }
+    }
+
+    private void createTempChannel(String name, String label) {
+        ChannelTypeUID temperature = new ChannelTypeUID("jablotron", "temperature");
+        ThingBuilder thingBuilder = editThing();
+        Channel channel = ChannelBuilder.create(new ChannelUID(thing.getUID(), name), "Number:Temperature")
+                .withLabel(label).withType(temperature).build();
+        thingBuilder.withChannel(channel);
+        updateThing(thingBuilder.build());
+    }
```

Comment:
Is there a reason why you create the temperature Channel dynamically? Is the temperature sensor not always present at JA-100 devices?

Reply:
It makes sense only to create it dynamically, it is not included out of the box, but as an optional peripheral.